File: MB-Secretaria/observability_py/review_metrics.py
from .types import ObservabilityEvent, ReviewMetrics


def calculate_review_metrics(events: list[ObservabilityEvent]) -> ReviewMetrics:
    scoped = [event for event in events if event.layer == "review"]
    return ReviewMetrics(
        pending_reviews=len([event for event in scoped if event.payload_summary.get("status") == "PENDING_REVIEW"]),
        escalation_count=len([event for event in scoped if "escalation" in event.event_type]),
        override_count=len([event for event in scoped if "override" in event.event_type]),
        blocked_outputs=len([event for event in scoped if "blocked" in event.event_type]),
        specialist_required_count=len([event for event in scoped if event.payload_summary.get("status") == "REQUIRES_SPECIALIST"]),
    )

File: MB-Chat/observability/review-metrics.ts
import type { ObservabilityEvent, ReviewMetrics } from "./types";

export function calculateReviewMetrics(events: ObservabilityEvent[]): ReviewMetrics {
  const scoped = events.filter((event) => event.layer === "review");
  return {
    pending_reviews: scoped.filter((event) => event.payload_summary.status === "PENDING_REVIEW").length,
    escalation_count: scoped.filter((event) => event.event_type.includes("escalation")).length,
    override_count: scoped.filter((event) => event.event_type.includes("override")).length,
    blocked_outputs: scoped.filter((event) => event.event_type.includes("blocked")).length,
    specialist_required_count: scoped.filter((event) => event.payload_summary.status === "REQUIRES_SPECIALIST").length,
  };
}
